R/forward_geocoding.R

```diff
@@ -51,6 +51,58 @@ oc_forward <-
+#' @param key Your OpenCage key.
+#' @param bounds Provides the geocoder with a hint to the region that the query resides in. This value will restrict the possible results to the supplied region. The bounds parameter should be specified as 4 coordinate points forming the south-west and north-east corners of a bounding box. For example, \code{bounds = c(-0.563160, 51.280430, 0.278970, 51.683979)} (min long, min lat, max long, max lat).
+#' @param countrycode Restricts the results to the given country. The country code is a two letter code as defined by the ISO 3166-1 Alpha 2 standard. E.g. "GB" for the United Kingdom, "FR" for France, "US" for United States.
+#' @param language An IETF format language code (such as "es" for Spanish or "pt-BR" for Brazilian Portuguese). If no language is explicitly specified, we will look for an HTTP Accept-Language header like those sent by a browser and use the first language specified and if none are specified "en" (English) will be assumed.
+#' @param limit How many results should be returned (1-100). Default is 10.
+#' @param min_confidence An integer from 1-10. Only results with at least this confidence will be returned.
+#' @param no_annotations Logical (default FALSE), when TRUE the output will not contain annotations.
+#' @param no_dedupe Logical (default FALSE), when TRUE the output will not be deduplicated.
+#' @param no_record Logical (default FALSE), when TRUE no log entry of the query is created at OpenCage.
+#' @param abbrv Logical (default FALSE), when TRUE addresses are abbreviated (e.g. C. instead of Calle)
+#' @param add_request Logical (default TRUE), when FALSE the query text is removed from the results data frame.
+#' @export
+
+oc_forward_vector <- function(x,
+                       key = oc_key(),
+                       bounds = NULL,
+                       countrycode = NULL,
+                       language = NULL,
+                       limit = 10,
+                       min_confidence = NULL,
+                       no_annotations = FALSE,
+                       no_dedupe = FALSE,
+                       no_record = FALSE,
+                       abbrv = FALSE,
+                       add_request = TRUE) {
+  if (is.data.frame(x) == TRUE) {
+    placename <- x[[1]]
+  } else {
+    placename <- x
+  }
+  results_df <- purrr::map(placename, opencage_forward,
+                           key = oc_key(),
+                           bounds = bounds,
+                           countrycode = countrycode,
+                           language = language,
+                           limit = limit,
+                           min_confidence = min_confidence,
+                           no_annotations = no_annotations,
+                           no_dedupe = no_dedupe,
+                           no_record = no_record,
+                           abbrv = abbrv,
+                           add_request = add_request)
+  results_df <- purrr::map(results_df, 1)
```

Comment:
You only take the first result for each placename. It's better to `limit` the API call then to `1` already.

Reply:
This actually just selects the first list of the output, which is the results data frame. It is equivalent to `results_df[[1]]`. This is something that would not be necessary if we changed the output in the whole `lapply` workflow or just returned the data frame of results instead of the list of four things.